lib/courses/widgets/course_detail_screen.dart

```diff
@@ -0,0 +1,89 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:schulcloud/app/services.dart';
+import 'package:schulcloud/app/widgets/app_bar.dart';
+import 'package:schulcloud/courses/bloc.dart';
+import 'package:schulcloud/courses/data/lesson.dart';
+import 'package:schulcloud/courses/widgets/lesson_screen.dart';
+
+class CourseDetailScreen extends StatelessWidget {
+  final Course course;
+
+  CourseDetailScreen({this.course});
+
+  @override
+  Widget build(BuildContext context) {
+    return ProxyProvider<ApiService, Bloc>(
+      builder: (_, api, __) => Bloc(api: api),
+      child: Scaffold(
+        appBar: AppBar(
+          iconTheme: IconThemeData(color: Colors.black),
+          title: Text(
+            course.name,
+            style: TextStyle(color: Colors.black),
+          ),
+          backgroundColor: course.color,
+        ),
+        bottomNavigationBar: MyAppBar(),
+        body: LessonList(
+          course: course,
+        ),
+      ),
+    );
+  }
+}
+
+class LessonList extends StatelessWidget {
+  final Course course;
+
+  const LessonList({this.course});
+
+  @override
+  Widget build(BuildContext context) {
+    return StreamBuilder<List<Lesson>>(
+      stream: Provider.of<Bloc>(context).getLessons(course.id),
+      builder: (context, snapshot) {
+        if (!snapshot.hasData) {
+          return Center(
+            child: CircularProgressIndicator(),
+          );
+        }
+
+        var tiles = [
+          Padding(
+            padding: const EdgeInsets.only(
+              top: 18,
+              bottom: 18,
+              left: 12,
+              right: 12,
+            ),
+            child: Text(
+              course.description,
+              style: TextStyle(fontSize: 20),
+            ),
+          ),
+          for (var lesson in snapshot.data)
+            ListTile(
+              title: Text(
+                lesson.name,
+                style: TextStyle(fontSize: 20),
+              ),
+              onTap: () => Navigator.push(
```

Comment:
Better create a helper method above that does this. Otherwise, this gets pretty deeply nested 🌴